archive_updater.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import os
+
+MAX_NUMBER_OF_RECENT_COMMITS = 2
+    
+def fill_message_directory(archive_path: str, directory_path: str, last_used_commit_hash: str) -> str:
+    '''Given a git repo path, directory path, and last used commit hash, this method updates the git repo,
+    then retrieves the MAX_NUMBER_OF_RECENT_COMMITS recent commits and converts them into files stored in
+    the directory corresponding to the passed in directory path. Throws CalledProcessError and Exception'''
+    
+    fetch = subprocess.check_call(['git', '-C', archive_path, 'fetch'])
+    
+    log = subprocess.Popen(
+        ['git', '-C', archive_path, 'log', '--format=format:%H', '-n',str(MAX_NUMBER_OF_RECENT_COMMITS)],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    
+    output, error = log.communicate()
+    if error:
+        raise Exception(f'Failed to log recent commits, see error: {error.decode()}')
+    message_hashes = output.decode('utf-8').split()
+    
+    if len(message_hashes) == 0:
+        raise Exception('There are no recent commits on the given repo')
+    
+    for i, hash in enumerate(message_hashes):
+        if hash == last_used_commit_hash:
+            break
+        file = os.path.join(directory_path, f'email{i}.txt')
```

Comment:
This email{i} naming convention will cause emails to be improperly indexed. First, the lesser the number, the more recent the commit (i.e. the most recent commit will be email0.txt). This can be changed so that it is reverse indexed (i.e. the greater the commit, the more recent) but email11.txt will be indexed before email2.txt. How should I adjust the code for this?

Reply:
Since email1.txt name already isn't in itself useful (you can find specific messages from it), why not use the SHA?
I.e. `f'email_{hash}.txt`

Those should never change.
But as noted elsewhere, long-term, I don't think we'll be creating files for all of these since the git repo already contains them in a more compact format and we could always run `git show {hash}:m` if we wanted to pull the email contents on demand.

Reply:
but the hashes won't necessarily be sorted right? As in a hash corresponding to a more recent commit will not necessarily appear before the hash of a later commit in a alphabetical sort, right? The current indexing system reads the files in alphabetical order (i think) and so files named with alphabetically smaller names will be indexed first, but they might be messages that are more recent and are a reply to a file that has yet to be indexed.

Reply:
Yes, you're probably getting the files in alphabetical order from `os.listdir()`
But you'd get `email1.text`, `email10.txt`, etc. and then `email2.txt`.

So relying on getting them in chronological order from `os.listdir()` isn't viable once we go past 9 emails (and I hope we do!)

We're going to need the indexer to be able to handle emails out of order.

Reply:
Seems like we've agreed that this loop can process emails out of order.

Can you go and update `file` to not use `i` in the name before we forget about it? :)